src/net.cpp

```diff
@@ -177,7 +173,8 @@ bool GetLocal(CService& addr, const CNode& peer)
             }
         }
     }
-    return nBestScore >= 0;
+    if (nBestScore == -1) return std::nullopt;
+    return addr;
```

Comment:
I think this can be simplified further?
```diff
diff --git a/src/net.cpp b/src/net.cpp
index 0d6b4f04a..c6c4c9a3e 100644
--- a/src/net.cpp
+++ b/src/net.cpp
@@ -149,7 +149,7 @@ uint16_t GetListenPort()
 [[nodiscard]] static std::optional<CService> GetLocal(const CNode& peer)
 {
     if (!fListen) return std::nullopt;
-    CService addr;
+    std::optional<CService> addr;
     int nBestScore = -1;
     int nBestReachability = -1;
     {
@@ -167,13 +167,12 @@ uint16_t GetListenPort()
             int nReachability = entry.first.GetReachabilityFrom(peer.addr);
             if (nReachability > nBestReachability || (nReachability == nBestReachability && nScore > nBestScore))
             {
-                addr = CService(entry.first, entry.second.nPort);
+                addr.emplace(CService(entry.first, entry.second.nPort));
                 nBestReachability = nReachability;
                 nBestScore = nScore;
             }
         }
     }
-    if (nBestScore == -1) return std::nullopt;
     return addr;
 }
 
```